src/Appwrite/Platform/Modules/Databases/Http/Databases/Collections/Update.php

```diff
@@ -104,7 +105,8 @@ public function action(string $databaseId, string $collectionId, string $name, ?
                 ->setAttribute('search', \implode(' ', [$collectionId, $name]))
         );
 
-        $dbForProject->updateCollection('database_' . $database->getSequence() . '_collection_' . $collection->getSequence(), $permissions, $documentSecurity);
+        $dbForDatabases = $getDatabasesDB($database);
+        $dbForDatabases->updateCollection('database_' . $database->getSequence() . '_collection_' . $collection->getSequence(), $permissions, $documentSecurity);
```

Comment:
Do we need to do this on dbForProject as well, since we have state in both DB for collections?